src/JustSaying/Messaging/Channels/Multiplexer/IMultiplexer.cs

```diff
@@ -0,0 +1,33 @@
+using System.Collections.Generic;
+using System.Threading;
+using System.Threading.Channels;
+using System.Threading.Tasks;
+using JustSaying.Messaging.Channels.Context;
+using JustSaying.Messaging.Interrogation;
+
+namespace JustSaying.Messaging.Channels.Multiplexer
+{
+    /// <summary>
+    /// Reduces multiple input streams of messages into a single stream, interleaving them.
+    /// </summary>
+    internal interface IMultiplexer : IInterrogable
+    {
+        /// <summary>
+        /// Begins reading from the ChannelReaders passed in from <see cref="ReadFrom"/>
```

Comment:
If you hover over the message it says `/// Begins reading from the ChannelReaders passed in from IMultiplexer.ReadFrom(ChannelReader<IQueueMessageContext>)`
So that might not be required?

Reply:
Have updated these docs slightly to include type info where possible